packages/test/end-to-end-tests/src/test/deRehydrateContainerTests.spec.ts

```diff
@@ -400,4 +400,113 @@ describe(`Dehydrate Rehydrate Container Test`, () => {
         assert.strictEqual(JSON.stringify(sharedMap3.snapshot()), JSON.stringify(sharedMapFromRC.snapshot()),
             "Snapshot of shared string should match and contents should be same!!");
     });
+
+    it("Container rehydration with not bounded dataStore handle stored in root of other bounded dataStore",
```

Comment:
I only diagonally scanned test cases, but it sounds we do not cover case where unbound DDS & data store (and not referenced anywhere) would be dropped while serializing. It would be nice to have that test case, even though it's less important (as there is no data loss, just perf)

Reply:
Added another test for this.